File: scala/parsers/src/test/scala/jp/co/teruuu/parser/json/JBoolParserTest.scala
package jp.co.teruuu.parser.json

import jp.co.teruuu.parser.common.{ParseNg, ParseOk}
import jp.co.teruuu.parser.json.`type`.JBool
import org.scalatest.FlatSpec

class JBoolParserTest extends FlatSpec{

  "parse" should "successt" in {
    val parser = JBoolParser

    val result1 = parser.parse("true")
    assert(result1.isInstanceOf[ParseOk[JBool]])
    assert(result1.asInstanceOf[ParseOk[JBool]].value.equals(JBool(true)))

    val result2 = parser.parse("false")
    assert(result2.isInstanceOf[ParseOk[JBool]])
    assert(result2.asInstanceOf[ParseOk[JBool]].value.equals(JBool(false)))
  }

  "parse" should "fail" in {
    val parser = JBoolParser

    val result1 = parser.parse("True")
    assert(result1.isInstanceOf[ParseNg[JBool]])
  }
}

File: java/parsers_java8/src/test/java/jp/co/teruuu/parser/json/JBoolParserTest.java
package jp.co.teruuu.parser.json;

import jp.co.teruuu.parser.common.ParseResult;
import jp.co.teruuu.parser.json.type.JBoolean;
import jp.co.teruuu.parser.json.type.JString;
import org.junit.Test;

public class JBoolParserTest {

    @Test
    public void test() {
        JParser parser = new JBoolParser();

        ParseResult result1 = parser.parse("true");
        assert(result1 instanceof ParseResult.Success);
        assert((ParseResult.Success) result1).value.equals(new JBoolean(true));

        ParseResult result2 = parser.parse("false");
        assert(result2 instanceof ParseResult.Success);
        assert((ParseResult.Success) result2).value.equals(new JBoolean(false));

        ParseResult result3 = parser.parse("Talse");
        assert(result3 instanceof ParseResult.Failure);
    }
}
